incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +291,31 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
+func (r *ObjectReconciler) ensureCanPropagate(nsNm string, inst *unstructured.Unstructured) bool {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return false
+	}
+	if inst.GetDeletionGracePeriodSeconds() != nil {
```

Comment:
I don't think this will compile - this doesn't return a pointer so it can't be compared to `nil`. See https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/controllers/hierarchy_controller.go#L116 for the correct way to check this.